test/unit/sentiment-analysis-client.spec.ts

```diff
@@ -0,0 +1,206 @@
+import { SentimentAnalysisClient, SentimentAnalysisJob, SentimentAnalysisJobOptions, SentimentAnalysisResult, SentimentAnalysisResultOptions } from '../../src';
+import { ApiRequestHandler } from '../../src/api-request-handler';
+import { GetListOfJobsOptions } from '../../src/models/GetListOfJobsOptions';
+
+jest.mock('../../src/api-request-handler');
+
+describe('sentiment-analysis-client', () => {
+    let sut: SentimentAnalysisClient;
+    let mockMakeApiRequest: jest.Mock;
+
+    const jobId = 'Umx5c6F7pH7r';
+    const otherJobId = 'EMx5c67p3dr';
+    const jobDetails = {
+        id: jobId,
+        status: 'completed',
+        created_on: '2018-05-05T23:23:22.29Z',
+        type: 'sentiment_analysis'
+    } as SentimentAnalysisJob;
+
+    beforeEach(() => {
+        mockMakeApiRequest = jest.fn();
+        (ApiRequestHandler as jest.Mock<ApiRequestHandler>).mockImplementationOnce(() => ({
+            makeApiRequest: mockMakeApiRequest
+        }));
+        sut = new SentimentAnalysisClient('testtoken');
+    });
+
+    describe('getJobDetails', () => {
+        it('get job by id', async () => {
```

Comment:
I guess I should've made this comment for topic as well but ideally these tests would also cover error cases